### PR TITLE
Added post Typescript

### PR DESCRIPTION
**Pull request by Eric-Jackson**
Added post module and utils module.

### CODE REVIEW

File: resources/assets/typescript/post.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+     * 
+     * @param postToggles Elements used to toggle hiding/showing a post
+     * @param postDeletes Elements used to delete a post
+     */
+    constructor(protected postToggles: NodeListOf<Element> = document.querySelectorAll('.post__toggle'),
+                protected postDeletes: NodeListOf<Element> = document.querySelectorAll('.delete a')) {
+        const _this = this;
+
+        if (postToggles.length > 0) {
+            utils.forEach(postToggles, function(i, toggle) {
```
Comment (euantorano) [inline resources/assets/typescript/post.ts]: Does TS not support arrow function syntax? (`(i, toggle) => {}`). I can't remember off the top of my head. Not a big deal either way.

File: resources/assets/typescript/utils.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export default class Utils {
+    forEach(array, callback, scope) {
```
Comment (euantorano) [inline resources/assets/typescript/utils.ts]: Should this not be in a function? Maybe I've just never come across this syntax before...

File: tests/js/test_typescript.html [hunk 3]
```diff
@@ -3,10 +3,25 @@
...
+                            <li class="post__controls__item post__controls__item--delete">
+                                <a href="{{ url_route("topics.delete", {'slug': topic.slug, 'id': topic.id, 'postId': post.id}) }}" class="post__controls__item__link icon fa-trash-o" title="{{ trans('topic.delete') }}">
+                                    <span class="text">{{ trans('topic.delete') }}</span>
+                                </a>
+                            </li>
+                        </ul>
+                    </div>
+                </div>
+
+    <script type="text/javascript" src="../../public/assets/js/mybb.js"></script>
```
Comment (euantorano) [inline tests/js/test_typescript.html]: use the `asset()` function to load assets :) `{{ asset('assets/js/mybb.js') }}`.

File: resources/assets/typescript/utils.ts (hunk 2 above)
Comment (Eric-Jackson) [reply resources/assets/typescript/utils.ts]: It's just defining a class method, I've made it a public method so it's a little more clear.

File: resources/assets/typescript/post.ts (hunk 1 above)
Comment (Eric-Jackson) [reply resources/assets/typescript/post.ts]: It does, went ahead and swapped those out.

File: tests/js/test_typescript.html (hunk 3 above)
Comment (Eric-Jackson) [reply tests/js/test_typescript.html]: As I mentioned, this is just a test page set up outside of the Laravel scope, not sure who by but it existed before I hopped in.

File: resources/assets/typescript/utils.ts (hunk 2 above)
Comment (euantorano) [reply resources/assets/typescript/utils.ts]: Nice, thanks.

File: tests/js/test_typescript.html (hunk 3 above)
Comment (euantorano) [reply tests/js/test_typescript.html]: Yep, sorry completely missed that. Carry on!

**Pull request merged**